toc.md

```diff
@@ -428,77 +429,234 @@ An additional backlog of 936 hours of image processing jobs everyday, along with
 
 ![scalibility_workerholic](/images/scalibility_workerholic.png)
 
-Well, that's easy. Workerholic is already scalable because we use Redis as a central data store for your jobs. Workerholic will still need access to the source code of your application, but Workerholic is not tied to a specific instance of your application, so you can have distributed web servers or a single web server and Workerholic will still work just fine. Workerholic can do this because its workers only care about the queue they're polling from, which is centralized with Redis.
+Workerholic is reliable, it uses concurrency, parallelism to be efficient and it is a scalable BJP. How does it compare to a gold standard like Sidekiq? What other optimizations can we introduce?
 
 ### Optimizations
-Once we had a fairly featured solution, we decided to compare against Sidekiq.
+
+When building a BJP it is important to see how the current implementation compares to a leader in the field. We decided to benchmark Workerholic against sidekiq in order to get some context as to where Workerholic is at and if we could optimize its performance.
 
 #### Serialization
+
+On our first iteration, we found that there was a great difference between Workerholic and Sidekiq; ours took much longer both on the enqueuing side and the processing side:
+
 ![optimizations_serialization_benchmark_yaml](/images/optimizations_serialization_benchmark_yaml.png)
 
-On our first iteration, we found that there was a great difference between Workerholic and Sidekiq; ours took much longer both on the enqueuing side and the processing side. Why was that? We looked into Sidekiq and found that it was using JSON serialization while we were using YAML, and so we decided to change our code to use JSON and see if that was really where the bottleneck was.
+What was our bottleneck? Sidekiq uses a similar model to Workerholic: concurrency via the use of threads. The difference seemed too big, especially since Workerholic should be a lighter BJP with less features and edge cases covered.
+
+The first thing we noticed is that both the enqueuing side and processing side were slower. This gave us a major insight into realizing it was a bottleneck present on both end of the spectrum. We started wondering if maybe there could be a problem performance hit because of serialization. Indeed, we were using YAML while Sidekiq is using JSON.
```

Comment:
both ends of the spectrum. We started wondering if the way we serialize our data may be the culprit. Indeed, we were using YAML while Sidekiq was using JSON.